JoonasHeinonen/escape-from-zordoom-prison
# Feature/ranged combat

### Link to the ticket: [Ranged combat](https://trello.com/c/dG71y3Ub/36-ranged-combat)

https://user-images.githubusercontent.com/33086486/191082816-db5d9ea4-8755-4aef-b9e0-17bc9ece6614.mp4

**Changes:**
- Created sprite for Angela’s blaster, added placeholder for the weapons in `player.tscn`. 
- Optimized walking code.
- Added new input map for the shooting functionality.
- Reserve X-button for ranged combat. (_Melee and ranged combat were separated in the original series._) 
- Shooting functionality implemented.

**How to test:**
 - Verify that the blaster is functioning.
 - Verify it ejects projectiles.
 - Verify that the projectiles explode when in contact with solid surface.

**Documentation & sources:**
 - [Godot official documentation - Input examples, InputMap](https://docs.godotengine.org/en/stable/tutorials/inputs/input_examples.html#inputmap)
 - [Godot FPS Tutorial - Projectile Weapons
](https://www.youtube.com/watch?v=IDsoEAj5xG0&ab_channel=Garbaj)
  - [Godot Recipes: Animation States](https://www.youtube.com/watch?v=0bq2OIjHxk4&ab_channel=KidsCanCode)
  - [How to animate Sprite Sheets in Godot using AnimationPlayer (beginner tutorial)](https://www.youtube.com/watch?v=FEwE6myyz_I&ab_channel=DrewConley)
